src/backend/distributed/utils/background_jobs.c

```diff
@@ -985,6 +1289,8 @@ StoreArgumentsInDSM(char *database, char *username, char *command, int64 taskId)
 	shm_toc_estimate_keys(&e, CITUS_BACKGROUND_TASK_NKEYS);
 	Size segsize = shm_toc_estimate(&e);
 
+	/* attaches the segment into TopMemoryContext, else, we can get garbage segment after commit */
+	CurrentResourceOwner = ResourceOwnerCreate(NULL, "citus background job");
```

Comment:
Why did you move this CurrentResourceOwner assignment here from the CitusBackgroundTaskExecutor function? It's also missing the assignment to CurrentMemoryContext. Is that on purpose?

Reply:
It is deliberately NULL because "If CurrentResourceOwner is NULL, the segment remains attached until explicitly detached or the session ends."